apps/services/user-profile/src/app/user-profile/verification.service.ts

```diff
@@ -133,11 +137,21 @@ export class VerificationService {
       }
     }
 
+    if (verification.tries >= maxRetries) {
+      return {
+        message:
+          'Too many failed email verifications. Please restart verification.',
+        confirmed: false,
+      }
+    }
+
     if (confirmEmailDto.hash !== verification.hash) {
-      // TODO: Add tries?
+      await verification.increment({ tries: 1 })
```

Comment:
I didn't now about the `increment` method, it should solve our concurrency issue 🙌 
According to the [docs](https://sequelize.org/api/v6/class/src/model.js~model#instance-method-increment) you could do `await verification.increment('tries')`. And it documents that it returns the updated model so don't we need to do

```suggestion
      verification = await verification.increment('tries')
```

Reply:
Yes, you are right, I didn't even think about that. It should cover the concurrency issue.